File: it.unibo.eclipse.qak.robotMinds19/resources/itunibo/coap/AsynchListener.java
package itunibo.coap;

import org.eclipse.californium.core.CoapHandler;
import org.eclipse.californium.core.CoapResponse;

public class AsynchListener implements CoapHandler {
    @Override
    public void onLoad(CoapResponse response) {
        String content = response.getResponseText();
        System.out.println("AsynchListener onLoad: " + content);
    }

    @Override
    public void onError() {
        System.err.println("AsynchListener Error");
    }
}

File: it.unibo.eclipse.qak.planning19/resources/itunibo/coap/observer/AsynchListener.kt
package itunibo.coap.observer

import org.eclipse.californium.core.CoapHandler
import org.eclipse.californium.core.CoapResponse
import itunibo.outgui.outguiSupport
import java.awt.Color

object AsynchListener : CoapHandler {

	val outDev = outguiSupport.create("Resource Coap OBSERVER", Color.green)
	
	override fun onLoad(response: CoapResponse?) {
		val content = response!!.getResponseText()
		outguiSupport.output("AsynchListener receives: $content"  )
	}
	override fun onError() {
		outguiSupport.output("AsynchListener Error")
	}
}